api.go

```diff
@@ -8,26 +8,26 @@ import (
 
 	"github.com/maniksurtani/quotaservice/config"
 	"github.com/maniksurtani/quotaservice/logging"
-
-	pbconfig "github.com/maniksurtani/quotaservice/protos/config"
+	pb "github.com/maniksurtani/quotaservice/protos/config"
 )
 
 // The Server interface is what you get when you create a new quotaservice.
 type Server interface {
 	Start() (bool, error)
 	Stop() (bool, error)
 	SetLogger(logger logging.Logger)
-	ServeAdminConsole(*http.ServeMux, string, config.ConfigPersister, bool)
+	CreateBucketContainer(*pb.ServiceConfig)
+	ServeAdminConsole(*http.ServeMux, string, bool)
 	SetListener(listener Listener, eventQueueBufSize int)
 }
 
 // New creates a new quotaservice server.
-func New(config *pbconfig.ServiceConfig, bucketFactory BucketFactory, rpcEndpoints ...RpcEndpoint) Server {
+func New(bucketFactory BucketFactory, persister config.ConfigPersister, rpcEndpoints ...RpcEndpoint) Server {
```

Comment:
Then how about:

`func New(b BucketFactory, p config.Persister, c pb.ServiceConfig, e ...RpcEndpoints)`

and 

`func NewWithDefaultConfigs(b BucketFactory, e ...RpcEndpoints)`

which could call `New()` with appropriate defaults? Saves us having to do `Server.CreateBucketContainer()` too.